src/main/scala/com/softwaremill/realworld/articles/ArticlesRepository.scala

```diff
@@ -61,6 +61,32 @@ class ArticlesRepository(quill: Quill.Sqlite[SnakeCase]):
       .map(_.map(article))
   }
 
+  def listArticlesByFollowedUsers(
+      pagination: Pagination,
+      followerId: Int
+  ): IO[SQLException, List[ArticleData]] = {
+
+    run(for {
+      ar <- sql"""
+                     SELECT DISTINCT * FROM articles a
```

Comment:
do we have to run 4 queries, or can we somehow combine them?

Reply:
I tried to do it in a different way, separete this "select in select", but also I had to run more queries. I have no idea how to combine them or run less queries.

Reply:
I think it's doable for profile data and favorited. First select is already joining relevant tables. I'm not sure how readable the resulting sql would be. For tags I believe it's better to leave them as they are. @wydra98 we can try together tomorrow, if you want.

Reply:
@adamw can i merge this, because it generally works, the only thing to improve is one sql query, but it would be good to improve more queries than in this PR. So i think it's better not to block this PR and make a separate issue and PR to improve sql queries. If you agree, please approve ;)

Reply:
sure, this sounds reasonable - add a task then :)